module/filter.go

```diff
@@ -58,6 +58,39 @@ type DeliveryContext struct {
 	Opts map[string]string
 }
 
+// DeepCopy method creates a copy of the DeliveryContext structure, also
+// copying contents of the maps and slices.
+//
+// There are two exceptions, however:
+// - SrcAddr is not copied and copys field references original value.
+// - Slice/map/pointer values in cpy.Ctx are not copied.
+func (ctx *DeliveryContext) DeepCopy() *DeliveryContext {
+	cpy := *ctx
+	// There is no good way to copy net.Addr, but it should not be
+	// modified by anything anyway so we are safe.
+
+	cpy.Ctx = make(map[string]interface{})
```

Comment:
We could initialize the map with the correct capacity

Reply:
What a good day, I didn't know that it is possible to set the capacity for a map.